### PR TITLE
Hides the OOBE window

### PR DESCRIPTION
**Pull request by CarlosNihelton**
This adds the capability to find the WSLg window where the OOBE GUI is displayed and controlling its visibility while synchronizing with the splash application, to ensure a smooth transition.

For the console, upon restoration, its resized to match the splash size and placement, so it can stay nicely hidden by the splash until it disappears. Also, asking to hide the splash now blocks while it stays visible.  This prevents other windows from popping up in front of it and causing disruption to the user attention.

I wanted to do the very same with the WSLg window, but moving it programmatically causes a very strange behavior. I'll investigate first whether this has been fixed in the latest updates or not and report a bug if not. It behaves as if the window had moved but the internal widgets had not, so that the user would have to guess the right offset to be able to click a button from where she sees it. So, for now, just toggling the OOBE GUI visibility.

Here is a video recording of the complete experience:

https://user-images.githubusercontent.com/11138291/163055156-ed2971fc-7ed4-48a0-bda4-3d324ee5c979.mp4

I had to tweak the CI. For some reason not yet clear to me clang-tidy was crashing, but with some patiency and brute-forcing it to try again I could get the CI running through again.

```diff
-    proc = subprocess.run(cli)
+    for i in range(0, 9):
+        proc = subprocess.run(cli)
+        if proc.returncode < 100:
+            break
+        print("\n\nClang-Tidy exit code: {}\n\n".format(proc.returncode))
```

### CODE REVIEW

File: DistroLauncher/installer_policy.h [hunk 1]
```diff
@@ -109,8 +160,6 @@ namespace Oobe
         static HANDLE start_installer_async(const wchar_t* command,
```
[the remark this answers is not in the document]
Comment (CarlosNihelton) [reply DistroLauncher/installer_policy.h]: Changed the function to take one single parameter.

**Comment by CarlosNihelton** [comment]
> ⚠️ Please check out carefully the isses found by `clang-tidy` in the new code ⚠️ ⚠️ `clang-format` found formatting issues in the code submitted.⚠️ Make sure to run clang-format and update this pull request. (1/1)

CI is working again!

File: .github/workflows/clang_lint/msbuild/clang_tidy_runner.py [hunk 2]
```diff
@@ -66,7 +66,11 @@ def run(*kwargs):
     print("==== Running clang-tidy with the following arguments:")
     print(cli)
 
-    proc = subprocess.run(cli)
+    for i in range(0, 9):
```
Comment (didrocks) [inline .github/workflows/clang_lint/msbuild/clang_tidy_runner.py:69]: Not a big fan, but yeah, sounds like the best option for now.
You added some rationale in the PR description, can you copy/paste it here as I don’t think we will remember this in few weeks time?

File: DistroLauncher/console_service.h [hunk 3]
```diff
@@ -63,26 +63,28 @@ namespace Win32Utils
...
+            if (state == consoleState()) {
                 return;
+            }
             fflush(stderrStream);
             fflush(stdoutStream);
             SetStdHandle(nStderrHandle, state.stdErrHandle);
             SetStdHandle(nStdoutHandle, state.stdOutHandle);
-            auto res2 = _dup2(state.stdErrFileDescriptor, _fileno(stderrStream));
-            auto res = _dup2(state.stdOutFileDescriptor, _fileno(stdoutStream));
+            [[maybe_unused]] auto res2 = _dup2(state.stdErrFileDescriptor, _fileno(stderrStream));
```
Comment (didrocks) [inline DistroLauncher/console_service.h:73]: I learnt about `[[maybe_unused]]` today :)

**Review by didrocks — APPROVED** [review]
This looks ace and polished! Excellent work :)

I haven’t spot anything worrying me in that PR. However, there is just one comment that should be added. Feel free to merge it as soon you do that. Approving :)

File: .github/workflows/clang_lint/msbuild/clang_tidy_runner.py (hunk 2 above)
Comment (didrocks) [reply .github/workflows/clang_lint/msbuild/clang_tidy_runner.py:77]: Thanks :)

Comment (CarlosNihelton) [reply .github/workflows/clang_lint/msbuild/clang_tidy_runner.py:77]: Done.

**Comment by didrocks** [comment]
See the failing test: the previous step failed (it should be set -e to fail?) but unsure why, then there is no .yaml file with the result

**Comment by CarlosNihelton** [comment]
> See the failing test: the previous step failed (it should be set -e to fail?) but unsure why, then there is no .yaml file with the result

I need more research on this. At that time it crashed exaustively. This time it succeeded after crashing once. This is disturbing and embarassing.

**Pull request merged**